chart/templates/_helpers.yaml

```
@@ -37,56 +37,74 @@ If release name contains chart name it will be used as a full name.
 {{/* Standard Airflow environment variables */}}
 {{- define "standard_airflow_environment" }}
   # Hard Coded Airflow Envs
+  {{- if .Values.enableBuiltInSecretEnvVars.AIRFLOW_CORE_FERNET_KEY }}
   - name: AIRFLOW__CORE__FERNET_KEY
     valueFrom:
       secretKeyRef:
         name: {{ template "fernet_key_secret" . }}
         key: fernet-key
+  {{- end }}
+  {{- if .Values.enableBuiltInSecretEnvVars.AIRFLOW__CORE__SQL_ALCHEMY_CONN }}
   - name: AIRFLOW__CORE__SQL_ALCHEMY_CONN
     valueFrom:
       secretKeyRef:
         name: {{ template "airflow_metadata_secret" . }}
         key: connection
+  {{- end }}
+  {{- if .Values.enableBuiltInSecretEnvVars.AIRFLOW_CONN_AIRFLOW_DB }}
   - name: AIRFLOW_CONN_AIRFLOW_DB
     valueFrom:
       secretKeyRef:
         name: {{ template "airflow_metadata_secret" . }}
         key: connection
+  {{- end }}
+  {{- if .Values.enableBuiltInSecretEnvVars.AIRFLOW__WEBSERVER__SECRET_KEY }}
   - name: AIRFLOW__WEBSERVER__SECRET_KEY
     valueFrom:
       secretKeyRef:
         name: {{ template "webserver_secret_key_secret" . }}
         key: webserver-secret-key
+  {{- end }}
   {{- if or (eq .Values.executor "CeleryExecutor") (eq .Values.executor "CeleryKubernetesExecutor") }}
+    {{- if .Values.enableBuiltInSecretEnvVars.AIRFLOW__CELERY__CELERY_RESULT_BACKEND }}
   - name: AIRFLOW__CELERY__CELERY_RESULT_BACKEND
```

```suggestion
  # (Airflow <1.10.0 variant)
  - name: AIRFLOW__CELERY__CELERY_RESULT_BACKEND
```

Might be nice to include why this is included here as well.

---

Technically it's 1.10.* though :). <1.10.0 is kinda wrong :)

---

It was renamed from `celery_result_backend` to `result_backend` in 1.10.0, meaning the former is for <1.10.0, no?

https://github.com/apache/airflow/blob/main/UPDATING.md#celery-config

(1.10.* looks at the latter: https://github.com/apache/airflow/commit/30076f1e45c7db54b0ac4bc118617d9a00cb5b7d)

---

aAAAAAAAAAAH... Do we really support 1.9.* :) . Should we drop it completely instead ?

---

1.9 EOL  -> Aug 27, 2018 :D.

---

Yeah, I think it's safe to drop it. You could just not touch it here and drop in another PR if you want to keep them isolated.